pkg/diffs/diffs.go

```diff
@@ -0,0 +1,68 @@
+package diffs
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	prowconfig "k8s.io/test-infra/prow/config"
+)
+
+// Diffs holds the information where the differences can be found.
+type Diffs struct {
+	ConfigPath    string
+	JobConfigPath string
+}
+
+// NewDiffs creates a new Diffs{}
+func NewDiffs(configPath, jobConfigPath string) *Diffs {
+	return &Diffs{
+		ConfigPath:    configPath,
+		JobConfigPath: jobConfigPath,
+	}
+}
+
+// GetChangedPresubmits returns a mapping of repo to presubmits to execute.
+func (d *Diffs) GetChangedPresubmits() (map[string][]prowconfig.Presubmit, error) {
```

Comment:
This is somehow expecting the PR to be checked out in a different location from the current working directory. In what set-up is this true? Will this be true in a PR job? Why have we chosen this?